### PR TITLE
Run CI against Rails 7.2

### PR DESCRIPTION
**Pull request by ohbarye**
Hi, based on [Rails 7.2 release](https://rubyonrails.org/2024/8/10/Rails-7-2-0-has-been-released), this pull request makes CI run tests against Rails 7.2.

Main changes are:

- Use rspec-rails 6.x for Rails 6.1 or above to follow rspec-rails' policy
    - ref https://github.com/rspec/rspec-rails/blob/d0e322b20bb713b200b66d4a7cc21a272d0b4374/README.md#L11-L16
- Use sqlite3 2.0 or above to follow https://github.com/rails/rails/pull/51958
- [chore] Add /gemfiles/vendor/ to .gitignore
    - I needed to run `BUNDLE_GEMFILE=gemfiles/activerecord_7_2.gemfile bundle` to debug test failures but it creates files under `gemfiles/vendor`. So I decided to let it ignore them.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -39,6 +39,12 @@ jobs:
           - ruby: '3.2'
             activerecord: '7.1'
             gemfile: 'activerecord_7_1.gemfile'
+          - ruby: '3.1'
```
Comment (ohbarye) [inline .github/workflows/test.yml:42]: Rails 7.2 supports Ruby 3.1 as a minimum version.

https://guides.rubyonrails.org/7_2_release_notes.html

Comment (Envek) [reply .github/workflows/test.yml:42]: I believe that it is enough to test against latest stable version of Ruby

**Comment by Envek** [comment]
Thank you very much for your pull request!

**Comment by ohbarye** [comment]
Thanks for your review!

**Pull request merged**